buildSrc/src/main/groovy/com/google/firebase/gradle/UploadMeasurementsTask.groovy

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle
+
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+import org.gradle.api.DefaultTask
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Takes the size information created by {@link GenerateMeasurementTask} and uploads it to the
+ * database using the uploader tool.
+ *
+ * <p>The uploader tool is fetched from the Internet using a URL. This URL, and the path to the
+ * report to upload, must be given as properties to this task. This task also requires a project
+ * property, {@code database_config} for connecting to the database. The format of this file is
+ * dictated by the uploader tool.
+ */
+public class UploadMeasurementsTask extends DefaultTask {
+
+    /**
+     * The URL of the uploader tool.
+     *
+     * <p>This must be a valid URL as a {@link String}.
+     */
+    @Input
+    String uploader
+
+    /**
+     * The file to upload.
+     *
+     * <p>This file must exist prior to executing this task, but it may be created by other tasks
+     * provided they run first.
+     */
+    @InputFile
+    File report
```

Comment:
Seems like a good use case for Gradle File Property
https://docs.gradle.org/current/userguide/lazy_configuration.html

Reply:
I like it this way, because it's clear in the Gradle file where the configuration file is placed and fetched.